stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/mentions/MentionsListAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package io.getstream.chat.android.ui.mentions
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.ui.databinding.StreamItemMentionListBinding
+import io.getstream.chat.android.ui.mentions.MentionsListAdapter.MessagePreviewViewHolder
+import io.getstream.chat.android.ui.mentions.MentionsListView.MentionSelectedListener
+import io.getstream.chat.android.ui.messagepreview.MessagePreviewView
+
+public class MentionsListAdapter : ListAdapter<Message, MessagePreviewViewHolder>(MessageDiffCallback) {
+
+    private var mentionSelectedListener: MentionSelectedListener? = null
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MessagePreviewViewHolder {
+        return StreamItemMentionListBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+            .let { MessagePreviewViewHolder(it.root) }
+    }
+
+    override fun onBindViewHolder(holder: MessagePreviewViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+    public fun setMentionSelectedListener(mentionSelectedListener: MentionSelectedListener?) {
+        this.mentionSelectedListener = mentionSelectedListener
+    }
+
+    public inner class MessagePreviewViewHolder(
```

Comment:
What's the reason making VH as inner class? Does it suppose to be customizable?

Reply:
It has to be an `inner` class to have access to the containing adapter. This is used to call the listener that's set on the adapter.